pkg/controller/podautoscaler/horizontal_test.go

```diff
@@ -509,6 +549,57 @@ func TestScaleUp(t *testing.T) {
 	tc.runTest(t)
 }
 
+func TestScaleUpUnreadyLessScale(t *testing.T) {
+	tc := testCase{
+		minReplicas:          2,
+		maxReplicas:          6,
+		initialReplicas:      3,
+		desiredReplicas:      4,
+		CPUTarget:            30,
+		CPUCurrent:           60,
+		verifyCPUCurrent:     true,
+		reportedLevels:       []uint64{300, 500, 700},
+		reportedCPURequests:  []resource.Quantity{resource.MustParse("1.0"), resource.MustParse("1.0"), resource.MustParse("1.0")},
+		reportedPodReadiness: []api.ConditionStatus{api.ConditionFalse, api.ConditionTrue, api.ConditionTrue},
+		useMetricsApi:        true,
+	}
+	tc.runTest(t)
+}
+
+func TestScaleUpUnreadyNoScale(t *testing.T) {
+	tc := testCase{
+		minReplicas:          2,
+		maxReplicas:          6,
+		initialReplicas:      3,
+		desiredReplicas:      3,
+		CPUTarget:            30,
+		CPUCurrent:           30,
+		verifyCPUCurrent:     true,
+		reportedLevels:       []uint64{300, 500, 700},
+		reportedCPURequests:  []resource.Quantity{resource.MustParse("1.0"), resource.MustParse("1.0"), resource.MustParse("1.0")},
+		reportedPodReadiness: []api.ConditionStatus{api.ConditionTrue, api.ConditionFalse, api.ConditionFalse},
+		useMetricsApi:        true,
+	}
+	tc.runTest(t)
+}
+
+func TestScaleUpUnreadyNoScaleWouldScaleDown(t *testing.T) {
```

Comment:
Can you point me to a difference in the definitions? They seem to be the same...